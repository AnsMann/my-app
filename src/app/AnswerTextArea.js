import React, { useState } from 'react'
import styled from 'styled-components'

const Answer = styled.textarea`
  border: 1px solid #007fbf;
  border-radius: 10px;
  font-family: inherit;
  font-size: 1rem;
  padding: 15px 20px 0;
  text-align: justify;
  width: 100%;
`

export function AnswerTextArea({
  name,
  placeholder = 'Type here',
  value,
  rows = '5',
}) {
  const [inputValue, setInputValue] = useState('')
  return (
    <Answer
      onChange={event => setInputValue(event.target.value)}
      rows={rows}
      placeholder={placeholder}
      name={name}
      value={value || inputValue}
    />
  )
}
