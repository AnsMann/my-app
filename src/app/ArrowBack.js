import React from 'react'
import styled from 'styled-components'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faLongArrowAltLeft } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import PropTypes from 'prop-types'

library.add(faLongArrowAltLeft)

const Back = styled.div`
  align-items: center;
  background-color: white;
  border: solid 2px #007fbf;
  border-radius: 20px;
  color: #007fbf;
  display: flex;
  font-size: 3.5rem;
  height: 60px;
  justify-content: center;
  left: 13px;
  position: absolute;
  top: 85px;
  width: 80px;
  z-index: 50;
`

export function ArrowBack({ onBackClick, history }) {
  return (
    <Back onClick={() => onBackClick(history)}>
      <FontAwesomeIcon icon={faLongArrowAltLeft} />
    </Back>
  )
}

ArrowBack.propTypes = {
  onBackClick: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
}
