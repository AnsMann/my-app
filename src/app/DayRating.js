import React from 'react'
import styled from 'styled-components'

const StyledRatingInput = styled.div`
  display: flex;
  font-size: 1.5rem;
  justify-content: space-around;
  padding: 1px;
  input {
    display: none;
    margin-right: 15px;
    &:checked + span {
      border-color: #007fbf;
      &:after {
        transform: scale(1);
      }
    }
  }
  label {
    position: relative;
  }
  span {
    border: 2px solid #007fbf;
    border-radius: 50%;
    display: block;
    float: left;
    height: 20px;
    margin-right: 10px;
    position: relative;
    top: 5px;
    width: 20px;
    &:after {
      background: #007fbf;
      border-radius: 50%;
      content: '';
      height: 10px;
      left: 3px;
      position: absolute;
      top: 3px;
      transform: scale(0);
      transition: all 0.2s ease;
      width: 10px;
    }
  }
`

export function DayRating() {
  const ratingOptions = [
    {
      rating: '3',
      output: 'ðŸ˜ƒ',
    },
    {
      rating: '2',
      output: 'ðŸ˜¶',
    },
    {
      rating: '1',
      output: 'ðŸ˜”',
    },
  ]

  return (
    <StyledRatingInput>
      <RatingOptions options={ratingOptions} />
    </StyledRatingInput>
  )
}

function RatingOptions({ options }) {
  return options.map(option => (
    <label key={option.rating}>
      <input type="radio" name="dayrating" value={option.rating} />
      <span />
      {option.output}
    </label>
  ))
}
