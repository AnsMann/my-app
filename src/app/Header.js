import React from 'react'
import styled from 'styled-components'

const StyledHeader = styled.header`
  align-items: center;
  background: #007fbf;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  color: #ffffff;
  display: flex;
  font-size: 2rem;
  justify-content: center;
`

export function Header({ title }) {
  return <StyledHeader>{title}</StyledHeader>
}
