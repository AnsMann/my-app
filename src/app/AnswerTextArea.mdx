---
name: Textarea
---

import { Playground, Props } from 'docz'
import { AnswerTextArea } from './AnswerTextArea.js'

# Textarea

<Props of={AnswerTextArea} />

**This textarea is used to create or edit diary entries**

<Playground>
  <AnswerTextArea
    name={'name'}
    defaultValue={'defaultvalue instead of placeholder'}
  />
</Playground>
