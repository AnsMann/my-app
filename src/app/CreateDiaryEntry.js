import React from 'react'
import { Header } from './Header'
import { DiaryEntryForm } from './DiaryEntryForm'
import { DiaryLogo } from './DiaryLogo'

export function CreateDiaryEntryForm({ onFormSubmit, history }) {
  return (
    <>
      <Header title={'Create Diary Entries'} />
      <DiaryLogo />
      <DiaryEntryForm onFormSubmit={onFormSubmit} history={history} />
    </>
  )
}
