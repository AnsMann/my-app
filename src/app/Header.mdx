---
name: Header
---

import { Playground, Props } from 'docz'
import { Header } from './Header.js'

# Header

<Props of={Header} />

**Header with given title.**

<Playground>
  <Header title={'Header title'} />
</Playground>
