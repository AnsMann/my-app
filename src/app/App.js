import React, { useState, useEffect } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import styled from 'styled-components'
import GlobalStyles from '../misc/GlobalStyles'
import moment from 'moment'
import 'moment/locale/de'
import ScrollMemory from 'react-router-scroll-memory'

import { Footer } from './Footer'
import { DiaryEntriesList } from './DiaryEntriesList'
import { CreateDiaryEntry } from './CreateDiaryEntry'
import {
  getEntriesFromMongoDB,
  setLocalStorage,
  getLocalStorage,
} from './services'
import { DiaryEntryDetails } from './DiaryEntryDetails'
import { ShareDiaryEntry } from './ShareDiaryEntry'
import { findIndex } from './utils'
import { EditDiaryEntry } from './EditDiaryEntry'
import { Settings } from './Settings'
import { NoConnectionModal } from './NoConnectionModal'

moment.locale('de')

const Grid = styled.div`
  display: grid;
  grid-template-rows: 80px 1fr 80px;
  height: 100vh;
`

export default function App() {
  const [diaryEntries, setDiaryEntries] = useState(
    getLocalStorage('myDiary') || []
  )
  const [sendAnonymous, setSendAnonymous] = useState(
    getLocalStorage('sendAnonymous') || false
  )
  const [workOffline, setWorkOffline] = useState(
    getLocalStorage('workOffline') || false
  )
  const [isNoConnectionModalVisible, setIsNoConnectionModalVisible] = useState(
    false
  )

  useEffect(() => {
    async function fetchDiaryEntries() {
      const entries = await getEntriesFromMongoDB()
      entries.name
        ? setIsNoConnectionModalVisible(true)
        : setDiaryEntries(entries)
    }
    workOffline || fetchDiaryEntries()
  }, [])

  useEffect(() => setLocalStorage('sendAnonymous', sendAnonymous), [
    sendAnonymous,
  ])

  useEffect(() => setLocalStorage('workOffline', workOffline), [workOffline])

  useEffect(() => {
    workOffline && setLocalStorage('myDiary', diaryEntries)
  }, [diaryEntries, workOffline])

  function handleWorkOfflineCheckbox() {
    setWorkOffline(!workOffline)
  }

  function handleAnonymousCheckbox() {
    setSendAnonymous(!sendAnonymous)
  }

  function handleFormSubmit(newDiaryEntries, history) {
    setDiaryEntries(newDiaryEntries)
    history.push('/')
  }

  function handleBackClick(history) {
    history.goBack()
  }

  function handleSharedDiaryEntry(newDiaryEntries) {
    setDiaryEntries(newDiaryEntries)
  }

  function handleDeleteClick(id, history, entryToDeleteInDB = null) {
    const index = findIndex(id, diaryEntries)
    entryToDeleteInDB
      ? setDiaryEntries([
          ...diaryEntries.slice(0, index),
          entryToDeleteInDB,
          ...diaryEntries.slice(index + 1),
        ])
      : setDiaryEntries([
          ...diaryEntries.slice(0, index),
          ...diaryEntries.slice(index + 1),
        ])
    history.push('/')
  }

  function handleEditOnDetailsPage(newDiaryEntries) {
    setDiaryEntries(newDiaryEntries)
  }

  function handleSyncButtonClick(updatedEntries) {
    setLocalStorage('myDiary', updatedEntries)
    setDiaryEntries(updatedEntries)
  }
  function resetNoConnectionModal() {
    setIsNoConnectionModalVisible(false)
  }

  return (
    <Router>
      <GlobalStyles />
      {isNoConnectionModalVisible && (
        <NoConnectionModal resetModal={resetNoConnectionModal} />
      )}
      <Grid>
        <ScrollMemory elementID="diary" />
        <Route
          exact
          path="/"
          render={props => (
            <DiaryEntriesList
              diaryEntries={diaryEntries}
              history={props.history}
              onDeleteClick={handleDeleteClick}
              workOfflineStatus={workOffline}
            />
          )}
        />
        <Route
          exact
          path="/create"
          render={props => (
            <CreateDiaryEntry
              onFormSubmit={handleFormSubmit}
              history={props.history}
              diaryEntries={diaryEntries}
              workOfflineStatus={workOffline}
            />
          )}
        />
        <Route
          exact
          path="/entries/:id"
          render={props => (
            <DiaryEntryDetails
              diaryEntries={diaryEntries}
              onBackClick={handleBackClick}
              {...props}
              onEditDetails={handleEditOnDetailsPage}
              workOfflineStatus={workOffline}
            />
          )}
        />
        <Route
          exact
          path="/entries/:id/share"
          render={props => (
            <ShareDiaryEntry
              diaryEntries={diaryEntries}
              diaryID={props.match.params.id}
              history={props.history}
              onBackClick={handleBackClick}
              onShare={handleSharedDiaryEntry}
              sendAnonymous={sendAnonymous}
              workOfflineStatus={workOffline}
            />
          )}
        />
        <Route
          exact
          path="/entries/:id/edit"
          render={props => (
            <EditDiaryEntry
              diaryEntries={diaryEntries}
              diaryID={props.match.params.id}
              history={props.history}
              onFormSubmit={handleFormSubmit}
              workOfflineStatus={workOffline}
            />
          )}
        />
        <Route
          exact
          path="/settings"
          render={props => (
            <Settings
              history={props.history}
              anonymousCheckboxStatus={sendAnonymous}
              onAnonymousCheckboxClick={handleAnonymousCheckbox}
              onworkOfflineCheckboxClick={handleWorkOfflineCheckbox}
              workOfflineCheckboxStatus={workOffline}
              onSyncButtonClick={handleSyncButtonClick}
              diaryEntries={diaryEntries}
            />
          )}
        />
        <Footer />
      </Grid>
    </Router>
  )
}
