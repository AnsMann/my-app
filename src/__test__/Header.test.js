import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Header } from '../app/Header'

Enzyme.configure({ adapter: new Adapter() })

describe('Header with different title', () => {
  it('renders an element', () => {
    const component = renderer.create(<Header />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders an element with props title', () => {
    const component = renderer.create(<Header title={'test title'} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
