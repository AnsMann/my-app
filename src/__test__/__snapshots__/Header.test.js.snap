// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with different title renders an element 1`] = `
<header
  className="Header__StyledHeader-rax5po-0 eSFArN"
/>
`;

exports[`Header with different title renders an element with props title 1`] = `
<header
  className="Header__StyledHeader-rax5po-0 eSFArN"
>
  test title
</header>
`;
